Pull Request US 07

SilviaPabon opened the pull request:
Este pull request incluye las tareas de:
- Comunicación con el back-end para enviar las búsquedas del usuario y obtener la repuesta del servidor
- Mostrar los vídeos enviados como respuesta del servidor haciendo uso de las cards desarrolladas

Quedo al pendiente!

SilviaPabon committed:
first draft, adding template to show videos

Changes in index.html and script.js

SilviaPabon committed:
lil' update in css to make all cards have same size

PedroChaparro committed:
Update font types and add vite.config.js file

SilviaPabon committed:
draft of response with setInterval

SilviaPabon committed:
updating script.js

-se agregaron comentarios
-se realizaron algunas correcciones para cuando se obtiene un array vacío
-se realizó una modificación en loading para eliminar los vídeos anteriores en una nueva búsqueda

PedroChaparro commented on src/index.html:
```diff
@@ -36,257 +36,57 @@ <h1 class="header__title" translate="no">ALAKO</h1>
 							target="_blank"
 							referrerpolicy="no-referrer"
-							>Te enseño a usar gimp en 26 minutos - edición de imágenes</a
+							id="url"
```
Sugiero que este id se cambie por una clase (También los otros casos en los que aplique), ya que los ids [deberían ser únicos en lo posible](https://developer.mozilla.org/en-US/docs/Web/API/Element/id).

PedroChaparro commented on src/script.js:
```diff
@@ -8,3 +12,90 @@ icon.onclick = function () {
+	const dataKey = await responseKey.json();
+
+	//requesting videos
+	let intervalID = setInterval(async () => {
```
(Esto es sobre todo para ver qué opinan) Creo que este interval se podría guardar en una variable fuera de la función, para después poder accederlo y cancelarlo (`clearInterval()`). Puede ser útil en el siguiente escenario, pero tiene un contra: 

1. El usuario hace una búsqueda
2. No ha llegado el resultado de la búsqueda pero el usuario hace otra, por lo que habrían dos intervals ejecutándose al tiempo.

Del modo que se me ocurre se podría cancelar el primer interval, pero entonces en la API se quedaría el resultado de esa búsqueda por siempre esperando que en algún momento se contulte la ruta `/obtain`, a no ser que lo eliminemos de otra manera.

Woynert replied:
Algo así cómo hacer un endpoint en la API para cancelar un trabajo.

Woynert commented on src/script.js:
```diff
@@ -8,3 +12,90 @@ icon.onclick = function () {
+				videos.forEach((element) => {
+					const tags = element.tags.split(', ', 3);
+					const templateClone = template.content.cloneNode(true);
+					const templateCloneTitle = templateClone.getElementById('title');
```
Complementando a @PedroChaparro emplear `getElementsByClassName("class")[0]` en lugar de `getElementById("id")` en los elementos internos del template. Debido a que el metodo `getElementById` no existe en el tipo Element (devuelto por cloneNode).

Woynert commented on src/script.js:
```diff
@@ -8,3 +12,90 @@ icon.onclick = function () {
+				//generating cards from template with the data from the response
+				videos.forEach((element) => {
+					const tags = element.tags.split(', ', 3);
+					const templateClone = template.content.cloneNode(true);
```
Esto devuelve un documentFragment que no recibe eventos; utilizar `template.cloneNode(true)` (sin el `content`) que devuelve un Element y si puede recibir eventos.

SilviaPabon replied:
Se realizaron los cambios @Woynert y @PedroChaparro quedo al pendiente.

SilviaPabon committed:
changed requested part I

SilviaPabon committed:
deleting template structure

se eliminaron las estructuras template que dificultaban agregar eventos y se propone alternativa de solución

SilviaPabon committed:
lil' update in index.html deleting unnecesary stuff

Woynert reviewed (commented):
El botón de lupa parece no servir, es esto intencional?

SilviaPabon committed:
adding event to search-icon

SilviaPabon committed:
lil' update in eventlistener

PedroChaparro commented on src/script.js:
```diff
@@ -8,3 +16,132 @@ icon.onclick = function () {
+								<div class="video-image">
+									<!-- Video image -->
+									<div class="video-image-container">
+										<img class="img-video" src="${element.thumbnail}" alt="" />
```
El atributo alt de la imágen está vacío, no es extremadamente grave, pero ya que se deben cambiar otras cosas, puede agregarlo. Propongo usar `${elem.title} youtube thumbnail`.

PedroChaparro commented on src/script.js:
```diff
@@ -8,3 +16,132 @@ icon.onclick = function () {
+						`;
+				});
+			} else {
+				loading();
```
Estos `else{loading()}` hacen que las cards deban ser eliminadas y creadas de nuevo en algunos casos donde no es necesario. Propongo llamar al `loading()` solo al inicio del método, debajo del `preventDefault()`.

PedroChaparro commented on src/script.js:
```diff
@@ -8,3 +16,132 @@ icon.onclick = function () {
+	const dataKey = await responseKey.json();
+
+	//requesting videos
+	intervalID = setInterval(async () => {
```
Intente enviar dos búsquedas al tiempo (sin esperar que llegue el resultado de la primera para hacer la segunda). En mi caso llega la primera correctamente pero **se queda pidiendo el mismo uuid a la api de scala**, por lo que nunca pide la segunda búsqueda y se queda mostrando errores en consola.

[Ejemplo](https://youtu.be/KNQuou3A6Y0)

SilviaPabon committed:
adding changes requested

PedroChaparro committed:
Fix multiple queries problem

PedroChaparro reviewed (approved):
TODO: Refactor in different file.
